service/worker/workerdeployment/client.go

```diff
@@ -1544,3 +1551,12 @@ func (d *ClientImpl) RegisterWorkerInVersion(
 
 	return nil
 }
+
+func (d *ClientImpl) getSyncBatchSize() int32 {
+	defaultSyncBatchSize := int32(25)
+	if n, ok := testhooks.Get[int](d.testHooks, testhooks.TaskQueuesInDeploymentSyncBatchSize); ok && n > 0 {
+		// In production, the testhook would be set to 0 and never reach here!
+		defaultSyncBatchSize = int32(n)
+	}
+	return defaultSyncBatchSize
```

Comment:
nit: I think default is a misnomer here.

```suggestion
	syncBatchSize := int32(25)
	if n, ok := testhooks.Get[int](d.testHooks, testhooks.TaskQueuesInDeploymentSyncBatchSize); ok && n > 0 {
		// In production, the testhook would be set to 0 and never reach here!
		syncBatchSize = int32(n)
	}
	return syncBatchSize
```